cypress_buildspec.yaml

```diff
@@ -0,0 +1,44 @@
+version: 0.2
+env:
+  shell: bash
+  secrets-manager:
+    SONAR_TOKEN: arn:aws:secretsmanager:eu-central-1:911845998067:secret:SonarCloud-7370vM:token
+phases:
+  install:
+    runtime-versions:
+      nodejs: 16
+    commands:
+      - apt-get update
+      - apt-get -y upgrade
+      # install yarn
+      - echo Installing yarn...
+      - curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+      - echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+      - apt install --no-install-recommends yarn
+      # install cypress dependencies
+      - apt-get install -y libgtk2.0-0 libgtk-3-0 libgbm-dev libnotify-dev libgconf-2-4 libnss3 libxss1 libasound2 libxtst6 xauth xvfb
+  pre_build:
+    commands:
+      - echo launching yarn install in $SUB_PRJ_NAME...
+      - cd $SUB_PRJ_NAME
+      - yarn install --frozen-lockfile
+  build:
+    commands:
+      - BRANCH_NAME=$(echo $CODEBUILD_WEBHOOK_HEAD_REF | sed 's/refs\/heads\///'); export BRANCH_NAME=${BRANCH_NAME:=$GIT_DEFAULT_BRANCH}
+      - if echo $CODEBUILD_SOURCE_VERSION | grep -q ^pr/ ; then PR_NUM=$(echo $CODEBUILD_SOURCE_VERSION | sed 's/pr\///'); BRANCH_TARGET=$(echo $CODEBUILD_WEBHOOK_BASE_REF | sed 's/refs\/heads\///'); export PR_NUM BRANCH_TARGET; fi;
+      ## Running the development server
```

Comment:
@maurizioflauti from line 27 to line 29 we set and export some env variables but we don't use them within cypress tasks. I think we should remove these lines

Reply:
ok

Reply:
@ZorbaDimatteo using setTimeout instead of wait-on would imply having to "manually" tune the timer instead of "dynamically" wait for the server to be up and running. This could lead to either have a more likely to fail pipeline (if, for some reason, the server takes longer than we expect and we went for a restrictive timer) or to waste a lot of time keep waiting, without any good reason, while the server is already up and running (if we define a conservative timeout). In this case I would have no doubts that the advantages of having a service tapping on my shoulders when the startup is done are well worth the downsides of installing a new (dev)dependency.